### PR TITLE
Ms.no hang

### PR DESCRIPTION
**Pull request by mariano54**
Based on the experiments from:
https://github.com/Chia-Network/chia-blockchain/pull/13087

It seems to fix the two issues:
- (Maybe) Hanging on ubuntu (and maybe other platforms)
- The windows Worker crash (by not running that test on windows)

### CODE REVIEW

File: tests/core/test_farmer_harvester_rpc.py [hunk 1]
```diff
@@ -414,6 +416,9 @@ async def test_farmer_get_harvester_plots_endpoints(
     reverse: bool,
     expected_plot_count: int,
 ) -> None:
+    if platform.system() == "Windows":
+        # Windows has an issue on Github CI with crashing workers
+        return
```
Comment (altendky) [inline tests/core/test_farmer_harvester_rpc.py:421]: Decorate with `@pytest.mark.skipif(sys.platform == "win32", reason="avoiding crashes on windows until we fix this").  Or whatever message.

File: tests/core/test_farmer_harvester_rpc.py [hunk 2]
```diff
@@ -534,7 +540,8 @@ async def assert_added(path: Path) -> None:
         await harvester_rpc_client.fetch("add_plot_directory", {})
 
     root_path = harvester_service._node.root_path
-    test_path = Path(root_path / "test_path").resolve()
+    random_name = token_bytes(32).hex()
+    test_path = Path(root_path / f"test_path_{random_name}").resolve()
```
Comment (altendky) [inline tests/core/test_farmer_harvester_rpc.py:544]: `Path()`ing a `/` joined `Path` object is redundant.  Could just drop the `Path` though I would write it as:

```suggestion
    test_path = root_path.joinpath(f"test_path_{random_name}").resolve()
```

Though really if we want a temporary directory we shouldn't shove it into the root path at all.  `tmp_path` is a fixture for getting a temp dir for the test.  Then don't clean them up.  pytest manages that for you, albeit that it keeps a few runs around before deleting the old temporary directories.

File: tests/core/test_farmer_harvester_rpc.py [hunk 3]
```diff
@@ -1,10 +1,12 @@
 import dataclasses
 import logging
 import operator
+import sys
 import time
 from math import ceil
 from os import mkdir
 from pathlib import Path
+from secrets import token_bytes
```
Comment (altendky) [inline tests/core/test_farmer_harvester_rpc.py:9]:
```suggestion
```

**Review by altendky — APPROVED** [review]

**Comment by mariano54** [comment]
Please approve @emlowe @wjblanke

**Review by emlowe — APPROVED** [review]

**Pull request merged**